parsl/dataflow/dflow.py

```diff
@@ -370,10 +370,16 @@ def handle_app_update(self, task_id, future, memo_cbk=False):
         # If checkpointing is turned on, wiping app_fu is left to the checkpointing code
         # else we wipe it here.
         if self.checkpoint_mode is None:
-            self.tasks[task_id]['app_fu'] = None
-        self.tasks[task_id]['depends'] = []
+            self.wipe_task(task_id)
         return
 
+    def wipe_task(self, task_id):
+        """ Remove task with task_id from the internal tasks table
+        """
+        logger.warning("Wiping {}".format(task_id))
+        # del self.tasks[task_id]
+        self.tasks.pop(task_id, None)
```

Comment:
does `pop` vs `del` have different the behaviour?

Reply:
del raises keyerror if the key is missing, while pop does not when a default is provided.

Reply:
were KeyErrors really happening?

Reply:
Good point. That shouldn't really happen. Let me try roll it back and see if it still causes issues in CI